include/material/bingham.tcc

```diff
@@ -0,0 +1,110 @@
+//! Read material properties
+template <unsigned Tdim>
+void mpm::Bingham<Tdim>::properties(const Json& material_properties) {
+  try {
+    density_ = material_properties["density"].template get<double>();
+    youngs_modulus_ =
+        material_properties["youngs_modulus"].template get<double>();
+    poisson_ratio_ =
+        material_properties["poisson_ratio"].template get<double>();
+    tau0_ = material_properties["tau0"].template get<double>();
+    mu_ = material_properties["mu"].template get<double>();
+    critical_shear_rate_ =
+        material_properties["critical_shear_rate"].template get<double>();
+    properties_ = material_properties;
+    status_ = true;
+  } catch (std::exception& except) {
+    console_->error("Material parameter not set: {}\n", except.what());
+  }
+}
+
+//! Elastic tensor is not defined in Bingham model, throws an error
+template <unsigned Tdim>
+Eigen::Matrix<double, 6, 6> mpm::Bingham<Tdim>::elastic_tensor() {
+
+  throw std::runtime_error("Elastic tensor is not used for this material");
+
+  return Eigen::Matrix<double, 6, 6>::Zero();
+}
+
+//! Compute stress without a particle handle is undefined in the Bingham model,
+//! throws an error
+template <unsigned Tdim>
+Eigen::Matrix<double, 6, 1> mpm::Bingham<Tdim>::compute_stress(
+    const Vector6d& stress, const Vector6d& dstrain) {
+
+  throw std::runtime_error("Stress computation for this material is not valid");
+
+  return Vector6d::Zero();
+}
+
+//! Compute stress
+template <unsigned Tdim>
+Eigen::Matrix<double, 6, 1> mpm::Bingham<Tdim>::compute_stress(
+    const Vector6d& stress, const Vector6d& dstrain,
+    const ParticleBase<Tdim>* ptr) {
+
+  const unsigned phase = 0;
+  auto strain_rate = ptr->strain_rate(phase);
+
+  // Get defintion of D for Bingham
+  strain_rate.tail(3) *= 0.5;
+
+  // Determine accuracy of minimum critical shear rate
+  const double shear_rate_threshold = 1.0E-15;
+  if (critical_shear_rate_ < shear_rate_threshold)
+    critical_shear_rate_ = shear_rate_threshold;
+
+  // Checking yielding from strain rate vs critical yielding shear rate
```

Comment:
This comment line doesn't apply to what you are doing. Move comment to the line above which you use it.